src/proto/coordinator.proto

```diff
@@ -0,0 +1,64 @@
+// API for the client to interact with the coordinator.
+
+syntax = "proto3";
+
+package coordinator;
+
+import "src/proto/common.proto";
+
+message ClientTransactionIdentifier {
+  // Unique for this client.
+  string transaction_id = 1;
+}
+
+message CommitAtomicTransactionRequest {
+  // Used by the coordinator in case the client does not get the response and
+  // retries. Required.
+  ClientTransactionIdentifier id = 1;
```

Comment:
Maybe just `string transaction_id`?

Reply:
Also, I’m wondering if this transaction_id is assigned by the coordinator so the client shouldn’t pass it in.

Reply:
Changed to `string client_transaction_id` and made it more clear that it's distinct from the one assigned by the coordinator. It's only used as a temporary cache for the coordinator to avoid resending client transactions that it has already processed.